mne/time_frequency/tests/test_spectrum.py

```diff
@@ -260,6 +282,76 @@ def test_spectrum_to_data_frame(inst, request, evoked):
     assert_frame_equal(_pick_first, _pick_last)
 
 
+def _agg_helper(df, weights, group_cols):
+    """Aggregate complex multitaper spectrum after conversion to DataFrame."""
+    from pandas import Series
+
+    unagged_columns = df[group_cols].iloc[0].values.tolist()
+    x_mt = df.drop(columns=group_cols).values[np.newaxis].T
+    psd = _psd_from_mt(x_mt, weights)
+    psd = np.atleast_1d(np.squeeze(psd)).tolist()
+    _df = dict(zip(df.columns, unagged_columns + psd))
+    return Series(_df)
+
+
+@pytest.mark.parametrize("long_format", (False, True))
+@pytest.mark.parametrize(
+    "method, output",
+    [("welch", "complex"), ("welch", "power"), ("multitaper", "complex")],
+)
+def test_unaggregated_spectrum_to_data_frame(raw, long_format, method, output):
+    """Test converting unaggregated spectra (multiple segments/tapers) to data frame."""
+    pytest.importorskip("pandas")
+    from pandas.testing import assert_frame_equal
+
+    from mne.utils.dataframe import _inplace
+
+    # aggregated spectrum → dataframe
+    orig_df = raw.compute_psd(method=method).to_data_frame(long_format=long_format)
+    # unaggregated welch or complex multitaper →
+    #   aggregate w/ pandas (to make sure we did reshaping right)
+    kwargs = dict()
+    if method == "welch":
+        kwargs.update(average=False)
+    spectrum = raw.compute_psd(method=method, output=output, **kwargs)
+    df = spectrum.to_data_frame(long_format=long_format)
+    grouping_cols = ["freq"]
+    drop_cols = ["segment"] if method == "welch" else ["taper"]
+    if long_format:
+        grouping_cols.append("channel")
+        drop_cols.append("ch_type")
+        orig_df.drop(columns="ch_type", inplace=True)
+    # only do a couple freq bins, otherwise test takes forever for multitaper
+    subset = partial(np.isin, test_elements=spectrum.freqs[:2])
+    df = df.loc[subset(df["freq"])]
+    orig_df = orig_df.loc[subset(orig_df["freq"])]
+    # sort orig_df, because at present we can't actually prevent pandas from
+    # sorting at the agg step *sigh*
+    _inplace(orig_df, "sort_values", by=grouping_cols, ignore_index=True)
+    # aggregate
+    df = df.drop(columns=drop_cols)
+    gb = df.groupby(grouping_cols, as_index=False, observed=False)
+    if method == "welch":
+        if output == "complex":
+
+            def _fun(x):
+                return np.nanmean(np.real(x * np.conj(x)))
```

Comment:
@tsbinns can you add a comment why we need `nanmean` here?

Reply:
I just took this from how the test was before, so I'm not sure why `nanmean` was chosen specifically (regular `mean` also works for all the test variations), however some kind of aggregation over observations is required.

Otherwise the subsequent call to the `agg()` method of the dataframe throws an error.

Reply:
https://github.com/mne-tools/mne-python/blob/361db716396adfbe4ea0a2f9b837bc7ccd132dd4/mne/time_frequency/tests/test_spectrum.py#L337-L338